.php-cs-fixer.php

```diff
@@ -3,46 +3,37 @@
 $finder = PhpCsFixer\Finder::create()
     ->in(__DIR__)
     ->name('*.php')
-    ->name('_ide_helper')
-    ->notName('*.blade.php')
+    // The next two rules are enabled by default, kept for clarity
     ->ignoreDotFiles(true)
     ->ignoreVCS(true)
-    ->notPath([
-        'cache',
-        'cypress',
-        'dbscripts',
-        'docs',
-        'js',
-        'lib/pkp',
-        'lib/ui-library',
-        'locale',
-        'node_modules',
-        'public',
-        'registry',
-        'schemas',
-        'styles',
-        'templates',
-    ]);
-
-// Apply formatting to all plugins that are not git submodules
-$pluginsDir = __DIR__ . '/plugins';
-$files = scandir($pluginsDir);
-foreach ($files as $file) {
-    $categoryDir = "{$pluginsDir}/{$file}";
-    if (!in_array($file, ['.', '..']) && is_dir($categoryDir)) {
-        $pluginDirs = scandir($categoryDir);
-        foreach ($pluginDirs as $pluginDir) {
-            $fullPluginPath = join('/', [$categoryDir, $pluginDir]);
-            $gitPath = join('/', [$fullPluginPath, '.git']);
-            if (!in_array($pluginDir, ['.', '..']) && is_dir($fullPluginPath) && file_exists($gitPath)) {
-                $finder->notPath(str_replace(__DIR__ . '/', '', $fullPluginPath));
-            }
+    // The pattern is matched against each found filename, thus:
+    // - The "/" is needed to avoid having "vendor" match "Newsvendor.php"
+    // - The presence of "node_modules" here doesn't prevent the Finder from recursing on it, so we merge these paths below at the "exclude()"
+    ->notPath($ignoredDirectories = ['cypress/', 'js/', 'locale/', 'node_modules/', 'styles/', 'templates/', 'vendor/'])
+    // Ignore root based directories
+    ->exclude(array_merge($ignoredDirectories, ['cache', 'dbscripts', 'docs', 'lib', 'public', 'registry', 'schemas']))
+    // Ignores Git folders
+    ->notPath((function () {
+        $recursiveIterator = new RecursiveIteratorIterator(
+            new RecursiveDirectoryIterator(__DIR__ . '/plugins', FilesystemIterator::SKIP_DOTS | FilesystemIterator::UNIX_PATHS | FilesystemIterator::CURRENT_AS_FILEINFO),
+            RecursiveIteratorIterator::SELF_FIRST
+        );
+        $recursiveIterator->setMaxDepth(1);
+        $gitFolders = new CallbackFilterIterator(
+            $recursiveIterator,
+            fn (SplFileInfo $file) => $recursiveIterator->getDepth() === $recursiveIterator->getMaxDepth()
+                && $file->isDir()
+                // Covers submodules (.git file) and external repositories (.git directory)
+                && file_exists("{$file}/.git")
```

Comment:
The main difference from the previous code (besides the comment I left on [pkp-lib](https://github.com/pkp/pkp-lib/pull/8990/files#r1191976990)) is that it also skips plugins that have a `.git` folder.